### PR TITLE
add tests for notebook

### PR DESCRIPTION
**Pull request by nastra**

### CODE REVIEW

File: colab/test_nessie_iceberg_demo_nba.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
+import pytest
```
Comment (snazy) [inline colab/test_nessie_iceberg_demo_nba.py:1]: Can you move the test-class into a separate directory?

File: colab/test_nessie_iceberg_demo_nba.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+
+@pytest.fixture(scope='module')
+def notebook():
+    with testbook('nessie-iceberg-demo-nba.ipynb', execute=True) as tb:
+        yield tb
+
+
+def test_notebook_output(notebook):
+    assert_that(notebook.cell_output_text(2)).contains(
```
Comment (snazy) [inline colab/test_nessie_iceberg_demo_nba.py:51]: No clue about testbook... so maybe my question is dumb:
Doesn't a cell needs to be executed before you can access the cell's output?
Would it help to delete all cell outputs before the test starts?

**Comment by snazy** [comment]
Guess, would also be nice to run it in CI - i.e. in the GH PR+Main workflows.

File: colab/test_nessie_iceberg_demo_nba.py (hunk 2 above)
Comment (nastra) [reply colab/test_nessie_iceberg_demo_nba.py]: the entire thing is being executed once via
```
@pytest.fixture(scope='module')
def notebook():
    with testbook('../../colab/nessie-iceberg-demo-nba.ipynb', execute=True) as tb:
        yield tb
```
due to the nature of the notebook (you can run most things exactly once and in the given order) this seems to be the best option. I was trying to test separate steps in separate test methods, but that's a bit difficult :(

File: colab/test_nessie_iceberg_demo_nba.py (hunk 1 above)
Comment (nastra) [reply colab/test_nessie_iceberg_demo_nba.py]: was thinking to move it to the same dir where `test_demo.py` is located, wdyt?

[the remark this answers is not in the document]
Comment (nastra) [reply colab/test_nessie_iceberg_demo_nba.py]: done

File: colab/test_nessie_iceberg_demo_nba.py (hunk 2 above)
Comment (snazy) [reply colab/test_nessie_iceberg_demo_nba.py]: Fair enough

File: colab/test_nessie_iceberg_demo_nba.py (hunk 1 above)
Comment (snazy) [reply colab/test_nessie_iceberg_demo_nba.py]: The stuff in `nessiedemo/` is rather a self-contained package/library. Having the tests for the notebooks in the tests for the library would introduce a somewhat cyclic dependency.

File: setup/requirements_dev.txt [hunk 3]
```diff
@@ -24,6 +25,7 @@ pytest-mypy==0.8.1
 pytest-runner==5.3.0
 Sphinx==3.5.4
 sphinx-autodoc-typehints==1.12.0
+testbook==0.4.1
```
Comment (snazy) [inline setup/requirements_dev.txt:28]: guess, this change can go away

File: colab/test_nessie_iceberg_demo_nba.py (hunk 1 above)
Comment (nastra) [reply colab/test_nessie_iceberg_demo_nba.py]: agreed. moved into its own directory

File: setup/requirements_dev.txt (hunk 3 above)
Comment (nastra) [reply setup/requirements_dev.txt]: fixed

File: .github/workflows/notebooks.yaml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+on: [push, pull_request]
+
+jobs:
+  python:
+    name: Testing Jupyter Notebooks
+    runs-on: ubuntu-latest
+    strategy:
+      max-parallel: 4
+      matrix:
+        python-version: [ 3.8 ]
```
Comment (snazy) [inline .github/workflows/notebooks.yaml:28]: Let's use 3.7, 3.8, 3.9. Most Notebook environments use 3.7 though.

File: .github/workflows/notebooks.yaml [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+name: Testing Jupyter Notebooks
+
+on: [push, pull_request]
+
+jobs:
+  python:
```
Comment (snazy) [inline .github/workflows/notebooks.yaml:22]: Mind moving this whole job into the pr and main workflows?

File: colab/nessie-iceberg-demo-nba.ipynb [hunk 6]
```diff
@@ -2020,7 +2020,7 @@
     }
    ],
    "source": [
-    "spark.sql(\"select count(*) from nessie.nba.salaries\").toPandas()"
+    "spark.sql(\"select count(*) from nessie.nba.salaries\").show()"
```
Comment (snazy) [inline colab/nessie-iceberg-demo-nba.ipynb:2023]: For a nicer representation, I'd prefer `toPandas`, but that has issues on Colab. Think, we need to figure out how to fix that (not at the moment, but later)

File: colab/tests/tox.ini [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+# limitations under the License.
+#
+
+[tox]
+envlist = py38
+skipsdist = True
+
+[gh-actions]
+python =
+    3.8: py38
```
Comment (snazy) [inline colab/tests/tox.ini:23]: same as the GH-workflow, let's keep 3.7, 3.8, 39

File: colab/tests/tox.ini [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+[tox]
+envlist = py38
```
Comment (snazy) [inline colab/tests/tox.ini:18]: same as the GH-workflow, let's keep 3.7, 3.8, 39

**Review by snazy — APPROVED** [review]
LGTM!

**Pull request merged**